worlds/_bizhawk/context.py

```diff
@@ -252,6 +254,15 @@ async def main():
         parser.add_argument("patch_file", default="", type=str, nargs="?", help="Path to an Archipelago patch file")
         args = parser.parse_args(launch_args)
 
+        if args.patch_file.startswith("archipelago://"):
+            url = urllib.parse.urlparse(args.patch_file)
+            args.connect = url.netloc
+            if url.username:
+                args.name = urllib.parse.unquote(url.username)
+            if url.password:
+                args.password = urllib.parse.unquote(url.password)
+            args.patch_file = ""
```

Comment:
This should probably just change the line under this to an elif instead of modifying the arg